widgetssdk/src/main/java/com/glia/widgets/callvisualizer/ActivityWatcherForCallVisualizer.kt

```diff
@@ -1,79 +1,60 @@
 package com.glia.widgets.callvisualizer
 
 import android.app.Activity
-import android.app.Activity.RESULT_OK
 import android.app.Application
-import android.content.Context
-import android.content.DialogInterface
 import android.content.Intent
-import android.media.projection.MediaProjectionManager
 import android.net.Uri
 import android.os.Bundle
 import android.provider.Settings
-import android.view.View
 import android.widget.Toast
 import androidx.activity.ComponentActivity
 import androidx.activity.result.ActivityResultLauncher
 import androidx.activity.result.contract.ActivityResultContracts
-import androidx.annotation.VisibleForTesting
 import androidx.appcompat.app.AlertDialog
 import com.glia.androidsdk.Glia
-import com.glia.androidsdk.GliaException
 import com.glia.androidsdk.comms.MediaDirection
 import com.glia.widgets.GliaWidgets
 import com.glia.widgets.R
 import com.glia.widgets.UiTheme
 import com.glia.widgets.call.CallActivity
 import com.glia.widgets.call.Configuration
-import com.glia.widgets.callvisualizer.controller.CallVisualizerController
-import com.glia.widgets.chat.ChatView
-import com.glia.widgets.core.dialog.Dialog
 import com.glia.widgets.core.dialog.DialogController
 import com.glia.widgets.core.dialog.model.DialogState
 import com.glia.widgets.core.notification.device.NotificationManager
-import com.glia.widgets.core.screensharing.ScreenSharingController
-import com.glia.widgets.core.screensharing.data.GliaScreenSharingRepository.SKIP_ASKING_SCREEN_SHARING_PERMISSION_RESULT_CODE
+import com.glia.widgets.core.screensharing.data.GliaScreenSharingRepository
 import com.glia.widgets.di.Dependencies
-import com.glia.widgets.filepreview.ui.FilePreviewView
 import com.glia.widgets.helper.Logger
 import com.glia.widgets.helper.Utils
 import com.glia.widgets.view.Dialogs
-import com.glia.widgets.view.head.controller.ServiceChatHeadController
-import com.google.android.material.theme.overlay.MaterialThemeOverlay
+import com.glia.widgets.view.unifiedui.exstensions.prepareContextWithStyle
 import java.lang.ref.WeakReference
 
 class ActivityWatcherForCallVisualizer(
-    private val callVisualizerController: CallVisualizerController,
-    private val screenSharingController: ScreenSharingController,
     private val dialogController: DialogController,
-    private var serviceChatHeadController: ServiceChatHeadController,
```

Comment:
@Pelkar Could you please explain the idea on the following topic? What is the reason why you left `DialogController` in `ActivityWatcherForCallVisualizer` and moved the other three controllers to `ActivityWatcherController`?

Reply:
My idea was to remove as many UI references from the controller as possible with the least amount of back and fourth between the watcher and the controller. Dialog controller uses a lot of view references and that is why I decided to keep that in the watcher side, because if we wanted to do this through the controller, then a lot of back and fourth is required to keep context out of the controller. That is why I extracted the dialog state handling to the controller, but the direct view handling was left in the watcher

A fair point can be made about screen sharing controller being similar in this regard and I agree. That is why all suggestions are welcome on this topic.